File: Trees/0404. Sum of Left Leaves/0404.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

 #pragma GCC optimize('03','unroll-loops')

class Solution {
public:
    int dfs(TreeNode* node){
        if(!node){
            return 0;
        }
        int sm = dfs(node->left) + dfs(node->right);
        if(node->left && !node->left->left && !node->left->right){
            sm+=node->left->val;
        }
        return sm;
    }

    int sumOfLeftLeaves(TreeNode* root) {
        ios_base::sync_with_stdio(0);
        cin.tie(0);
        cout.tie(0);

        return dfs(root);
    }
};

File: Trees/0404. Sum of Left Leaves/0404.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        def dfs(node):
            if not node:
                return 0
            sm = dfs(node.left) + dfs(node.right)
            if node.left and not node.left.left and not node.left.right:
                sm += node.left.val
            return sm
        return dfs(root)